ftlib/__init__.py

```diff
@@ -42,17 +74,31 @@ def unlock(self):
         self.lock_count = self.lock_count - 1
         logging.debug("after unlock, lock count: {}".format(self.lock_count))
 
-    def initialized(self):
-        return self._initialized
+    def init(
+        self,
+        consensus,
+        framework,
+        consensus_init_kwargs=None,
+        framework_init_kwargs=None,
+    ):
+        assert framework in framework_list
+        assert consensus in consensus_list
+
+        if consensus == "shared_storage":
+            from .consensus.shared_storage import SharedStorage
```

Comment:
Move all import statements to top of the file

Reply:
Even if I move these import statements to the top of the file with a try clause, it will still import modules that will never be used.

The contemporary design keeps the memory usage minimal. If it is just for coding style, I prefer keeping the the import statements here.

Is there any other reason to move the import statements?

Reply:
One possible reason is to fail fast.

Reply:
> One possible reason is to fail fast.

@chunyang-wen Do you suggest at the top of the file, we testing availability of all modules that might be imported later? I can give a try and see if we can take such tests without importing them (to keep the memory usage minimal).